src/main/java/hudson/plugins/jira/JiraMailAddressResolver.java

```diff
@@ -32,9 +34,17 @@ public String findMailAddressFor(User u) {
         }
         String username = u.getId();
 
-        Job<?, ?> job = Stapler.getCurrentRequest().findAncestorObject( Job.class);
+        Job<?, ?> job = null;
+        try {
+            job = Stapler.getCurrentRequest().findAncestorObject(Job.class);
+        } catch (NullPointerException e) {
+            // sadly but outside of a context job this throw an exception
```

Comment:
Since there is not test added... what is the usecase (i.e. what are the circumstances under which this error is thrown?)

Reply:
I can try to add some tests but not sure how. there is no much test currently for this part.
Just create your own SecurityListener extension (https://javadoc.jenkins.io/jenkins/security/SecurityListener.html)
just after login on the home page the method authenticated2 is fired and you're not part of any job context.

this make me think we could add some caching for the getUser method from Jira? nothing really urgent but why not.

Reply:
> I understand that you're able to reproduce this error, so... that's the test (or scenario) I was asking about :-) There is already a test for the [resolver](https://github.com/jenkinsci/jira-plugin/blob/master/src/test/java/hudson/plugins/jira/MailResolverTest.java) that is exactly checking the null in case it was disabled via properties. This NPE is different though, so I'm trying to understand why this resolver is fired when you only log in..? I'm happy with the fix itself though.

Just create your own SecurityListener extension (https://javadoc.jenkins.io/jenkins/security/SecurityListener.html) and in try to figure out the logged in user email with such code 
```
            User user = User.getById(userId, false);
            if (user != null) {
                Mailer.UserProperty emailProperty = user.getProperty(Mailer.UserProperty.class);
                if (emailProperty != null && emailProperty.getAddress() != null) {
                    email = emailProperty.getAddress();
                }
            }
```
trying to get the user email will fire this NPE. 
For some reason in this case (I haven't really dig into it but I presume it's because we are in the context of spring security handlers and not yet stapler) 
```
StaplerRequest req = Stapler.getCurrentRequest();
here req is null 
```

> 
> Adding caching might be good as a separate feature, not as a fix to this scenario, because it wouldn't fix it, only make it harder to debug.

yes definitely it was just loud thinking :) and as I said nothing urgent

Reply:
@rantoniuk  test added